android-app-examples/SNetDemo/app/src/main/java/com/example/snetdemo/ImageSegmentationActivity.java

```diff
@@ -0,0 +1,449 @@
+
+
+        PackageManager pm = this.getPackageManager();
+        if (!pm.hasSystemFeature(PackageManager.FEATURE_CAMERA_ANY))
+        {
+            mIsDeviceWithCamera = false;
+            btn_GrabCameraImage.setEnabled(false);
+        }
+
+        channelID = this.getResources().getInteger(R.integer.channel_id);
+
+        AsyncTask task = new AsyncTask<Object, Object, Object>()
+        {
+
+            protected void onPreExecute()
+            {
+                super.onPreExecute();
+
+                textViewProgress.setVisibility(View.VISIBLE);
+                textViewProgress.setText("Opening service channel");
+                loadingPanel.setVisibility(View.VISIBLE);
+
+                getWindow().setFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE,
+                        WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE);
+
+
+            }
+
+            @Override
+            protected Object doInBackground(Object... param)
+            {
+                SnetSdk sdk = new SnetSdk(ImageSegmentationActivity.this);
+                semanticSegmentationService = new SemanticSegmentationService(sdk, BigInteger.valueOf(channelID));
+                return null;
+            }
+
+            protected void onPostExecute(Object obj)
+            {
+                textViewProgress.setVisibility(View.INVISIBLE);
+                loadingPanel.setVisibility(View.INVISIBLE);
+                getWindow().clearFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE);
+            }
+
+
+        };
+        task.execute();
+
+    }
+
+    public void sendRunImageSegmentationMessage(View view)
+    {
+        if(this.mIsInputImageUploaded)
+        {
+            AsyncTask task = new AsyncTask<Object, Integer, Object>()
+            {
+                protected void onPreExecute()
+                {
+                    super.onPreExecute();
+                    textViewProgress.setVisibility(View.VISIBLE);
+                    loadingPanel.setVisibility(View.VISIBLE);
+
+                    btn_UploadImageInput.setEnabled(false);
+                    btn_RunImageSegmentation.setEnabled(false);
+
+                    btn_GrabCameraImage.setEnabled(false);
+
+                    getWindow().setFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE,
+                            WindowManager.LayoutParams.FLAG_NOT_TOUCHABLE);
+
+
+                }
+
+                protected Void doInBackground(Object... param)
+                {
+                    publishProgress(new Integer(PROGRESS_LOADING_IMAGE));
+                    Bitmap bitmap = null;
+                    try
+                    {
+                        bitmap = handleSamplingAndRotationBitmap(ImageSegmentationActivity.this, Uri.fromFile(new File(mImageInputPath)));
+                    }
+                    catch (IOException e)
+                    {
+                        Log.e("ERROR IN LOADING BITMAP ", Calendar.getInstance().getTime().toString() + e.toString());
```

Comment:
To log errors you can just use the following call, and remove `e.printStackTrace()`:
```suggestion
                        Log.e("ImageSegmentationActivity", "ERROR IN LOADING BITMAP, timestamp: " + Calendar.getInstance().getTime().toString(), e);
```
First parameter is `tag` which should give log reader understanding of the location of the error if I understand correctly (https://developer.android.com/reference/android/util/Log). Documentation recommends use String constant here. Second parameter is error message, and in case of errors third parameter is exception which will be logged properly including stack trace.